# Copyright (C) 2019-2020 LEIDOS.

# Licensed under the Apache License, Version 2.0 (the "License"); you may not
# use this file except in compliance with the License. You may obtain a copy of
# the License at
 
# http://www.apache.org/licenses/LICENSE-2.0
 
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
# License for the specific language governing permissions and limitations under
# the License.
 
# Double: The length of the trajectory in time domain, in seconds
# Units: Seconds
trajectory_time_length: 6.0

# String: The default control plugin name
control_plugin_name: "pure_pursuit"


# Double : Minimum allowable speed
# Units : m/s
minimum_speed : 2.2352

# Double : Maximum allowable longitudinal acceleration 
# Units : m/s^2
max_accel : 1.5

# Double : Minimum value for lookahead distance 
# Units : meters
minimum_lookahead_distance : 5.0

# Double : Maximum value for lookahead distance 
# Units : meters
maximum_lookahead_distance : 25.0

# Double : Minimum speed value for lookahead calculation
# Units : m/s
minimum_lookahead_speed : 2.8

# Double : Maximum speed value for lookahead calculation
# Units : m/s
maximum_lookahead_speed : 13.9

# Double : Maximum allowable lateral acceleration 
# Units : m/s^2
lateral_accel_limit : 1.5

# Int : Size of the window used in the moving average filter to smooth both the computed curvature and output speeds
speed_moving_average_window_size : 5

#Int: // Size of the window used in the moving average filter to smooth the curvature profile
curvature_moving_average_window_size : 9

# Int : Number of points to look ahead when calculating the curvature of the lanelet centerline
curvature_calc_lookahead_count : 1

# Int: Amount to downsample input lanelet centerline data. Value corresponds to saving each nth point.
downsample_ratio : 1

#double : Acceptable range from the ending downtrack, when planning success can be published, if vehicle has changed lanes
#Units: meters
destination_range : 0

#Double : The time after which a lane change request can be considered to time out
#Units : seconds
lanechange_time_out: 6.0

# Double: The minimum timestep used for planning trajectory
#Units: seconds
min_timestep : 0.1

#Double:  The range from the start from which the plugin can start working
#Units: meters
starting_downtrack_range : 5.0

#Double: Fraction of the maneuver that has been completed. Starting fraction specifies a higher urgency of in the request to lane change
starting_fraction : 0.2

#Double: Fraction of the maneuver that has been completed. Mid fraction specifies a lower urgency in the request to lane change
mid_fraction : 0.5

#Double: The minimum gap required between 2 vehicles to start lane changing
#Units: meters
min_desired_gap: 5.0

# Double : The additional downtrack beyond requested end dist used to fit points along spline. Extra points removed before returning trajectory. Feature not used if end dist is route end
#Units: meters
ending_buffer_downtrack : 5.0
